lib/Sema/CSDiag.cpp

```diff
@@ -4264,8 +4264,14 @@ bool TypeChecker::diagnoseSelfAssignment(const Expr *E) {
   if (!AE)
     return false;
 
+  // Allow the use of parentheses to express self-assignment when it's a deliberate choice
+  auto src = AE->getSrc();
+  auto loadSrc = dyn_cast<LoadExpr>(src);
```

Comment:
Because this is checking for self-assignment, it wouldn't make sense to use `let` anyway, right? Checking just for `ParenExpr` won't work because of the cases where we do have a `LoadExpr`. Would I need to check for any of the "container" expressions here, instead of just the `LoadExpr`?